src/api/app/controllers/person/token_controller.rb

```diff
@@ -36,6 +36,30 @@ def delete
       render_ok
     end
 
+    # PUT /person/<login>/token/<id>
+    def update
+      authorize @user, :update?
+
+      permitted_params = params.slice(:enabled).permit(:enabled)
+
+      if permitted_params.blank?
+        render_error status: 400,
+                     errorcode: 'invalid_token_attributes',
+                     message: 'Parameters to modify token attributes are missing.'
+        return
+      end
+
+      unless %(true, false).include?(permitted_params[:enabled])
+        render_error status: 400,
+                     errorcode: 'invalid_token_attribute_value',
+                     message: "'#{permitted_params[:enabled]}' is not a valid 'enabled' attribute value. Use 'true' or 'false'."
+        return
+      end
+
+      @user.tokens.find(params[:id]).update(permitted_params)
+      render_ok
```

Comment:
This will always render okay, no matter if the above `update` succeeds or not.

Reply:
> This will always render okay, no matter if the above `update` succeeds or not.

If the above `update` throws an exception, it will not render okay.

If the above `update` doeesn't change the values of the token, it should render okay.

What do you mean exactly by "succeeds"?

Reply:
I changed `update` to `update!`. Was that what you meant?

Reply:
Applied.